src/Common/Middleware/AddXMsClientRequestIdMiddleware.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AzureOss\Storage\Common\Middleware;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * @internal
+ */
+class AddXMsClientRequestIdMiddleware
```

Comment:
Final since we do not want people outside the library to extend it since it will cause BCs